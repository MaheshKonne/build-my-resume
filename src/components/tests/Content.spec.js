import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import ContentComp from "../Content";

describe('<ContentComp />', () => {
    it('should match with snapshot with default values', () => {
        const component = renderer.create(<ContentComp>Hello</ContentComp>).toJSON();
        expect(component).toMatchSnapshot();
        expect(component).toHaveStyleRule('color', '#000');
        expect(component).toHaveStyleRule('letter-spacing', 'normal');
    });

    it('should match with snapshot with passed props', () => {
        const props = {
            color: 'red',
            letterSpacing: '10px'
        };
        const component = renderer.create(<ContentComp {...props}>Hello</ContentComp>).toJSON();
        expect(component).toMatchSnapshot();
        expect(component).toHaveStyleRule('color', 'red');
        expect(component).toHaveStyleRule('letter-spacing', '10px');
    });
});