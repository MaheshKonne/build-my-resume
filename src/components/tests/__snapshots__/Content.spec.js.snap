// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentComp /> should match with snapshot with default values 1`] = `
.c0 {
  margin: 0;
  color: #000;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

<p
  className="c0"
  color="#000"
  letterSpacing="normal"
>
  Hello
</p>
`;

exports[`<ContentComp /> should match with snapshot with passed props 1`] = `
.c0 {
  margin: 0;
  color: red;
  -webkit-letter-spacing: 10px;
  -moz-letter-spacing: 10px;
  -ms-letter-spacing: 10px;
  letter-spacing: 10px;
}

<p
  className="c0"
  color="red"
  letterSpacing="10px"
>
  Hello
</p>
`;
